ext/proj4_c_impl/main.c

```diff
@@ -362,6 +348,32 @@ static VALUE method_proj4_is_valid(VALUE self)
   return data->pj ? Qtrue : Qfalse;
 }
 
+static VALUE method_proj4_is_crs(VALUE self)
+{
+  RGeo_Proj4Data *self_data;
+  int i;
+  int proj_type;
+  int valid_types[] = {
+    PJ_TYPE_CRS, PJ_TYPE_GEODETIC_CRS, PJ_TYPE_GEOCENTRIC_CRS, PJ_TYPE_GEOGRAPHIC_CRS,
+    PJ_TYPE_GEOGRAPHIC_2D_CRS, PJ_TYPE_GEOGRAPHIC_3D_CRS, PJ_TYPE_VERTICAL_CRS,
+    PJ_TYPE_PROJECTED_CRS, PJ_TYPE_COMPOUND_CRS, PJ_TYPE_TEMPORAL_CRS,
+    PJ_TYPE_ENGINEERING_CRS, PJ_TYPE_BOUND_CRS, PJ_TYPE_OTHER_CRS
+  };
+  int valid_types_size = 13;
+
+  TypedData_Get_Struct(self, RGeo_Proj4Data, &rgeo_proj4_data_type, self_data);
+  if (self_data->pj){
+    proj_type = proj_get_type(self_data->pj);
+    for (i = 0; i < valid_types_size; i++) {
+      if(valid_types[i] == proj_type) {
+        return Qtrue;
+      }
+    }
+  }
+
+  return Qfalse;
+}
```

Comment:
I think we could have something much simpler here by using the [`proj_is_crs`](https://proj.org/development/reference/functions.html#c.proj_is_crs) function.

```c
proj_is_crs(self_data->pj) ? Qtrue : Qfalse
```

I think the null check is even already done by proj!

Reply:
Good find! I was looking for a function that handled that but missed it. Definitely much easier to just use that